File: Puntonet/Puntonet.Web/Modules/ServerTypes/Parameters/AtributtesRow.ts
﻿import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface AtributtesRow {
    IdAtributte?: number;
    Description?: string;
}

export abstract class AtributtesRow {
    static readonly idProperty = 'IdAtributte';
    static readonly nameProperty = 'Description';
    static readonly localTextPrefix = 'Parameters.Atributtes';
    static readonly lookupKey = 'Parameters.Atributtes';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<AtributtesRow>('Parameters.Atributtes') }
    static async getLookupAsync() { return getLookupAsync<AtributtesRow>('Parameters.Atributtes') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<AtributtesRow>();
}

File: Puntonet/Puntonet.Web/Modules/Parameters/Atributtes/AtributtesRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace Puntonet.Parameters
{
    [ConnectionKey("Default"), Module("Parameters"), TableName("[dbo].[Atributtes]")]
    [DisplayName("Atributtes"), InstanceName("Atributtes")]
    [ReadPermission("Administration:General")]
    [ModifyPermission("Administration:General")]
    public sealed class AtributtesRow : Row<AtributtesRow.RowFields>, IIdRow, INameRow
    {
        [DisplayName("Id Atributte"), Identity, IdProperty]
        public int? IdAtributte
        {
            get => fields.IdAtributte[this];
            set => fields.IdAtributte[this] = value;
        }

        [DisplayName("Description"), Size(200), NotNull, QuickSearch, NameProperty]
        public string Description
        {
            get => fields.Description[this];
            set => fields.Description[this] = value;
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field IdAtributte;
            public StringField Description;

        }
    }
}